subsys/net/ip/net_context.c

```diff
@@ -2817,6 +2821,21 @@ int net_context_set_option(struct net_context *context,
 		break;
 	}
 
+	/* Create a branch that just makes sure that the generic
+	 * setter function do not give build error even if all
+	 * options using the setter are disabled.
+	 */
+	if (0) {
+		uint16_t opt16;
+		uint8_t opt8;
+		bool optbool;
+		int optint;
+
+		(void)set_uint16_option(&opt16, &optint, sizeof(int));
+		(void)set_uint8_option(&opt8, &opt8, sizeof(uint8_t));
+		(void)set_bool_option(&optbool, &optint, sizeof(int));
+	}
+
```

Comment:
Why not add `__maybe_unused` to those functions?

Reply:
Hmm, why not indeed. I did not knew about such attribute, but will change the code to use that.